### PR TITLE
LPD-41686 AddToCart

### PR DESCRIPTION
**Pull request by smottal**
https://liferay.atlassian.net/browse/LPD-41686

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:report:74486006

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:report:76382552

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java [hunk 1]
```diff
@@ -1951,6 +1954,67 @@ private boolean _jsonMatches(String json1, String json2) {
 			return true;
 		}
 
+		try {
```
Comment (gianmarcobrunialti) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java:1957]: This is super-intensive but currently necessary to handle the different formats for items with options coming from Add to Cart or Quick Add to Cart. It'll be solved elsewhere via: [LPD-25497](https://liferay.atlassian.net/browse/LPD-25497)

**Review by gianmarcobrunialti — APPROVED** [review]

**Review by gianmarcobrunialti — APPROVED** [review]
This is good to go for the major performance issue at stake. The component may need a refactor, but only after the ticket indicated in the previous comment.

**Comment by smottal** [comment]
ci:report:78019846

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:report:79043949

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:79482483

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**